scripts/template_upload_rhevm.py

```diff
@@ -81,9 +81,16 @@ def change_edomain_state(api, state, edomain, provider):
             export_domain = dc.storagedomains.get(edomain)
             if export_domain:
                 if state == 'maintenance' and export_domain.get_status().state == 'active':
-                    dc.storagedomains.get(edomain).deactivate()
+                    # may be tasks on the storage, try multiple times
+                    print('RHEVM:{} {} in active, waiting for deactivate...').format(
+                        provider, edomain)
+                    wait_for(lambda: dc.storagedomains.get(edomain).deactivate(), delay=5,
+                             num_sec=600, handle_exception=True)
```

Comment:
This does have a negative side effect of burying the exceptions from the `activate` or `deactivate` call so we won't get tracebacks on them, but a timeout exception instead.